File: js/0066.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function (digits) {
    digits.reverse();

    var carry = 1;
    for (let i = 0; i < digits.length; i++) {
        digits[i] += carry;
        carry = 0;
        if (digits[i] >= 10) {
            digits[i] = digits[i] % 10;
            carry = 1;
        }
        if (carry == 0) break;
    }
    if (carry) digits.push(1);
    digits.reverse();
    return digits;
};

File: py/0066.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        digits = digits[::-1]
        carry = 1
        for i in range(len(digits)):
            digits[i] += carry
            carry = 0
            if digits[i] > 9:
                digits[i] = digits[i] % 10
                carry = 1
            if carry == 0:
                break
        if carry:
            digits.append(1)
        digits.reverse()
        return digits
